improved kafka actions reading (#1217)
citrusframework/citrus: src/manual/endpoint-kafka.adoc

          consumer-group="citrus_group"/>
----

Note that for effective message consumption it is advisable to use random consumer groups.
Both the Java and XML DSL support random consumer groups, when enabled: `randomConsumerGroup(true)` or `random-consumer-group="true"`.

The endpoint is now ready to be used inside a test case.
The test simply references the endpoint by its name when sending or receiving.

In case of a send operation the endpoint creates a Kafka producer and will simply publish the records to the defined Kafka topic.

> As the communication is asynchronous by default producer does not wait for a synchronous response.

Reviewer: ```suggestion
As the communication is asynchronous by the default producer, the producer does not wait for a synchronous response.
```

In case of a receive operation the endpoint creates a Kafka consumer instance in the defined (possibly random) consumer group.

NOTE: By default, the consumer starts a subscription on the given topic and acts as a polling listener.
This means that the message consumer connects to the given topic and polls one single record at a time.

> As soon as a record has been received the action is ready to perform the validation process that verifies the record.

Reviewer: ```suggestion
As soon as a record has been received, the action is ready to perform the validation process that verifies the record.
```

> This requires looping for finding specific Kafka messages and is potentially very ineffective, given the Kafka topic receives a high amount of traffic.

Reviewer: Würde den Satz ein bisschen anders gestalten.
```suggestion
This approach requires looping to find specific Kafka messages, which can be highly inefficient, especially when the Kafka topic experiences heavy traffic.
```

| consumer-group
| No
| `citrus_kafka_group`
| Consumer group name.
Please keep in mind that records are load balanced across consumer instances with the same consumer group name set.
So you might run into message timeouts when using multiple Kafka endpoints with the same consumer group name.

> | random-consumer-group

Reviewer: ```suggestion
| `random-consumer-group`
```

Author: I've corrected this for all properties, not only the random consumer group.
